approov-service/src/main/java/io/approov/service/okhttp/ApproovService.java

```diff
@@ -862,6 +903,69 @@ else if (approovResults.getStatus() != Approov.TokenFetchStatus.UNKNOWN_KEY)
             }
         }
 
+        // if message signing is enabled, add the signature header to the request
+        if (messageSigningConfigs != null) {
+            // Match the domain of the request URL against the configured message signing domains
+            String domain = request.url().host();
+            ApproovMessageSigningConfig messageSigningConfig = messageSigningConfigs.get(domain);
+            if (messageSigningConfig == null) {
+                // If no specific configuration is found for the domain, use the default configuration
+                domain = "*";
+                messageSigningConfig = messageSigningConfigs.get(domain);
+            }
+            if (messageSigningConfig == null) {
+                // If no default configuration is found, do not sign the message
+                return chain.proceed(request);
+            }
+            Log.d(TAG, "Signing message with configuration for domain: "  + domain + ":  " + (messageSigningConfig.getSignApproovToken() ? ", " + approovTokenHeader +  " **** " : "") + " headers " + messageSigningConfig.getSignedHeaders() +
+                    (messageSigningConfig.getSignBody() ? ", body" : ""));
+            // build the message to sign, consisting of the URL, the names and values of the included headers and
+            // the body, if enabled, where each entry is separated from the next by a newline character
+            StringBuilder message = new StringBuilder();
+            //  1. add the URL to the message, followed by a newline
+            message.append(request.url());
+            message.append("\n");
+            // 2. Add the Method to the message
+            message.append(request.method());
+            message.append("\n");
+            // 3. add the Approov token to the message, if required (TODO: Note we add the approov token and the prefix!?????)
+            if (messageSigningConfig.signApproovToken) {
+                if (approovResults.getToken() != null) {
+                    message.append(approovTokenPrefix + approovResults.getToken());
```

Comment:
You should read the approov header from the request instead of recreating the contents here - it is just a safer approach as the same process will occur on the server side -once you migrate the code, into the DefaultRequestSigningConfig you will have to do that anyway.